BackEndLib/Wchar.cpp

```diff
@@ -560,20 +560,27 @@ int _Wtoi(const WCHAR* wsz)
 }
 
 //*****************************************************************************
-WCHAR* _itoW(int value, WCHAR* wcz, int radix)
+WCHAR* _itoW(int value, WCHAR* wcz, int radix, int bufferLength)
 {
 	int i = 0;
 	bool bNegative = false;
 	if (value < 0) //negative?
 	{
 		value = -value;
 		bNegative = true;
+		bufferLength--; // We have one character fewers for numbers because we need negative sign
 	}
+	bufferLength--; // Make space for null termination char
 	do {
 		int val = value % radix;
 		if (val < 10)
 			buffer[i++] = (value % radix) + '0';
 		else buffer[i++] = (value % radix) - 10 + 'A';
+
+		if (i >= bufferLength) {
+			ASSERTP(i < bufferLength, "Converting a string to a number ran out of space in the provided buffer.");
```

Comment:
We don't need the condition check here, because it's always false.
Suggest:
ASSERT(!"_itoW: Converting a string to a number ran out of space in the provided buffer.");

Reply:
I completely forgot you can use `ASSERT()` like that, been a while since I touched this codebase :). Posted an update

Reply:
No worries!  Same here :-)